pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -431,24 +442,47 @@ func (l *listUsersQuery) expandExclusion(
 
 	baseFoundUsersMap := make(map[string]struct{}, 0)
 	for fu := range baseFoundUsersCh {
-		key := tuple.UserProtoToString(fu)
+		key := tuple.UserProtoToString(fu.user)
 		baseFoundUsersMap[key] = struct{}{}
 	}
 	subtractFoundUsersMap := make(map[string]struct{}, len(baseFoundUsersMap))
+	negatedSubtractionMap := make(map[string]struct{})
 	for fu := range subtractFoundUsersCh {
-		key := tuple.UserProtoToString(fu)
+		for _, s := range fu.subtracted {
+			negatedSubtractionMap[tuple.UserProtoToString(s)] = struct{}{}
+		}
+		key := tuple.UserProtoToString(fu.user)
 		subtractFoundUsersMap[key] = struct{}{}
```

Comment:
If you have `user:* but not user:maria` as an intermediate result, I don't think it is safe to separate them into separate maps.

For example, I believe this is why this test fails:

```go
tests := ListUsersTests{
		{
			name: "Mixed",
			model: `
			model
				schema 1.1
			type user
			type org
				relations
					define member: [user]
			type document
				relations
					define viewer: viewer1 or viewer2
					define viewer1: [org#member] but not blocked1
					define viewer2: [user:*] but not blocked2
					define blocked1: [user]
					define blocked2: [user]
		`,
			tuples: []*openfgav1.TupleKey{
				tuple.NewTupleKey("document:1", "viewer1", "org:fga#member"),
				tuple.NewTupleKey("org:fga", "member", "user:maria"),

				tuple.NewTupleKey("document:1", "viewer2", "user:*"),
				tuple.NewTupleKey("document:1", "blocked1", "user:will"),
			},
			req: &openfgav1.ListUsersRequest{
				Object:      &openfgav1.Object{Type: "document", Id: "1"},
				Relation:    "viewer",
				UserFilters: []*openfgav1.ListUsersFilter{{Type: "user"}},
			},
			expectedUsers: []string{"user:*", "user:maria", "user:will"},
		},
	}
```

Reply:
> If you have `user:* but not user:maria` as an intermediate result, I don't think it is safe to separate them into separate maps.

Can you elaborate on what this means?

That aside, `user:will` is not being returned because that user is not assigned to either `org:fga#member` or `document:1#viewer`. This works exactly as I would expect.

Reply:
> Can you elaborate on what this means?

i need to give this more thought but my gut reaction is that we should probably key just one map with `user:* but not user:maria`, not 1 map with key `user:*` and 1 map with key `user:maria`

Reply:
> That aside, user:will is not being returned because that user is not assigned to either org:fga#member or document:1#viewer. This works exactly as I would expect.

Because of the tuple `document:1 # viewer2 @ user:*`, user:will _should_ be returned. The corresponding Check request returns `allowed=true`: https://play.fga.dev/sandbox/?store=asd2

Reply:
Checking `user:foo`,`user:bar` and `user:baz` all return `allowed=true` too but I would not expect them to be returned from ListUsers.

Reply:
Oh... my assumption was that if we have a tuple mentioning the user, then it'd be somehow included in the response. I need to double-check other tests

Reply:
Welp, I tried to poke holes but couldn't 😄  nice work Will!